### PR TITLE
Ipv6 support

### LINKED ISSUE #12: Support for IPv6 subnets
First of all, thanks for making this. This is a really interesting project.
I was wondering how hard it would be to support IPv6-only subnets. By the looks of it, only https://github.com/skx/simple-vpn/blob/master/cmd_server.go#L170 assumes IPv4 and that's because we treat it like a string instead of like an IP address.
The use-case is when using simple-vpn within another private subnet, using IPv6-only makes it easy to pick an IP block that won't clash with the rest of the subnet.
Would you be able to support this use-case? Open to PRs?

### PR DESCRIPTION
**Pull request by skx**
This pull-request is a toy to add IPv6 support.

* The way addresses are assigned handles IPv6-addresses to clients and the server.
* The handling of packets is conditional on the IPv4 vs. IPv6 type.
* I've tested with Linux clients and hosts.
* It needs more testing, and the configuration/readme needs updated to mention IPv6.

Compare with #14 , this closes #12.

### CODE REVIEW

**Comment by skx** [comment]
I believe this is complete, tested, and working now:

* For IPv4
* For IPv6

Will merge tomorrow unless there are any surprising results.

**Comment by samridh90** [comment]
I'll test this out sometime today...
For my own curiosity, what would it take to get unicast to work fro IPv6? It wasn't obvious to me why v6 is harder to implement than v4.

**Comment by samridh90** [comment]
@skx : ☝️

**Comment by skx** [comment]
A detailed explanation would take longer time than I have to spare.

But you need to think about routing and IP-headers.  For IPv4 you get the MAC address of the source and the destination in the packet.  Using those you can lookup the appropriate socket and write the whole packet down the correct socket.

For IPv6 if you want to do it properly you need to do that, but you'd also want to handle other things such as router advertisements and neighbour solications.

In short think of IPv4 and think about how ARP requests are handled; here we cheat and don't handle them.  If you want to do "stuff properly" you should.  And that's whats missing for the IPv6 side.  (Only ARP isn't involved, instead it's neighbour solicatation and RA.)

Feel free to experiment in your fork.  It'll be fiddly and annoying, but also educational.
